runtime/vara/src/tests.rs

```diff
@@ -0,0 +1,100 @@
+        let interval_end: u32 = $interval_mid + $interval_mid / INTERVAL_ERROR_DIVIDER;
+
+        if !(interval_start <= weight_v && weight_v <= interval_end) {
+            let instr_name = stringify!($instruction);
+            println!("FAILED: Interval mismatch for {instr_name}");
+            println!("Weight is {weight_v} ps. Expected interval is [{interval_start}, {interval_end}] ps");
+            panic!();
+        }
+    };
+}
+
+#[test]
+fn heuristics_test() {
+    let instruction_weights = InstructionWeights::<crate::Runtime>::default();
+
+    check_weight_inbounds_interval!(instruction_weights, i64const, 283);
+    check_weight_inbounds_interval!(instruction_weights, i64load, 30_700);
+    check_weight_inbounds_interval!(instruction_weights, i64store, 42_600);
+    check_weight_inbounds_interval!(instruction_weights, select, 7_100);
+    check_weight_inbounds_interval!(instruction_weights, r#if, 6_000);
+    check_weight_inbounds_interval!(instruction_weights, br, 3_300);
+    check_weight_inbounds_interval!(instruction_weights, br_if, 6_000);
+    check_weight_inbounds_interval!(instruction_weights, br_table, 9_900);
+    check_weight_inbounds_interval!(instruction_weights, br_table_per_entry, 435);
+
+    check_weight_inbounds_interval!(instruction_weights, call, 4_900);
+    check_weight_inbounds_interval!(instruction_weights, call_indirect, 22_100);
+    check_weight_inbounds_interval!(instruction_weights, call_indirect_per_param, 1_700);
+
+    check_weight_inbounds_interval!(instruction_weights, local_get, 856);
+    check_weight_inbounds_interval!(instruction_weights, local_set, 1_940);
+    check_weight_inbounds_interval!(instruction_weights, local_tee, 2_000);
+    check_weight_inbounds_interval!(instruction_weights, global_get, 2_100);
+    check_weight_inbounds_interval!(instruction_weights, global_set, 2_900);
+    check_weight_inbounds_interval!(instruction_weights, memory_current, 14_200);
+
+    check_weight_inbounds_interval!(instruction_weights, i64clz, 6_100);
+    check_weight_inbounds_interval!(instruction_weights, i64ctz, 5_700);
+    check_weight_inbounds_interval!(instruction_weights, i64popcnt, 1_600);
+    check_weight_inbounds_interval!(instruction_weights, i64eqz, 3_400);
+    check_weight_inbounds_interval!(instruction_weights, i64extendsi32, 1_200);
+    check_weight_inbounds_interval!(instruction_weights, i64extendui32, 650);
+    check_weight_inbounds_interval!(instruction_weights, i32wrapi64, 375);
+    check_weight_inbounds_interval!(instruction_weights, i64eq, 3_200);
+    check_weight_inbounds_interval!(instruction_weights, i64ne, 3_200);
+
+    check_weight_inbounds_interval!(instruction_weights, i64lts, 3_200);
+    check_weight_inbounds_interval!(instruction_weights, i64ltu, 3_200);
+    check_weight_inbounds_interval!(instruction_weights, i64gts, 3_200);
+    check_weight_inbounds_interval!(instruction_weights, i64gtu, 3_200);
```

Comment:
Could we make one macros call for all of this instructions? They are all from one group. May be we can make macros with generic amount of arguments.

Reply:
It have no sense in current context. We have only a few same instructions, generally all instructions are different.